src/plugins/vis_type_vislib/public/vislib/lib/types/point_series.test.js

```diff
@@ -215,18 +217,30 @@ describe('Point Series Config Type Class Test Suite', function () {
   });
 
   describe('line chart', function () {
-    beforeEach(function () {
+    function prepareData(percentileTestdata) {
       const percentileDataObj = {
         get: (prop) => {
           return maxBucketData[prop] || maxBucketData.data[prop] || null;
         },
         getLabels: () => [],
         data: percentileTestdata.data,
       };
-      parsedConfig = vislibPointSeriesTypes.line(percentileTestdata.cfg, percentileDataObj);
-    });
+      return (parsedConfig = vislibPointSeriesTypes.line(
+        percentileTestdata.cfg,
+        percentileDataObj
+      ));
+    }
+
     it('should render a percentile line chart', function () {
+      const parsedConfig = prepareData(percentileTestdata);
       expect(JSON.stringify(parsedConfig)).toEqual(JSON.stringify(percentileTestdataResult));
     });
+
+    it('should render a percentile line chart when value is float', function () {
+      const parsedConfig = prepareData(percentileTestdataFloatValue);
+      expect(JSON.stringify(parsedConfig)).toEqual(
+        JSON.stringify(percentileTestdataFloatValueResult)
+      );
```

Comment:
you could also check like this 
```suggestion
       expect(parsedConfig).toMatchObject(percentileTestdataFloatValueResult);
```
(or check the specific value that caused the wrong rendering to be correct, less JSON involved )